plugin/storage/integration/kafka_test.go

```diff
@@ -88,6 +88,11 @@ func (s *KafkaIntegrationTestSuite) initialize(t *testing.T) {
 	s.SpanReader = &ingester{traceStore}
 	s.CleanUp = func(_ *testing.T) {}
 	s.SkipArchiveTest = true
+	t.Cleanup(func() {
+		require.NoError(t, f.Close())
+		require.NoError(t, spanConsumer.Close())
+		testutils.VerifyGoLeaksOnce(t)
+	})
```

Comment:
wrong place for VerifyGoLeaksOnce and overall a bad pattern. The pattern should be scheduling close of the object as soon as it is successfully constructed, not waiting till the end of the function (by which time it might exit due to other errors).

```
obj, err := NewObject()
require.NoError(t, err)
t.Cleanup(func() { assert.NoError(t, obj.Close()) })
```